### ISSUE TITLE
[Paper] Fixing the Double Penalty in Data-Driven Weather Forecasting Through a Modified Spherical Harmonic Loss Function

### ISSUE THREAD

**Issue by jacobbieker** [openclimatefix/graph_weather#135]
## Arxiv/Blog/Paper Link
https://arxiv.org/abs/2501.19374

## Detailed Description
Interesting idea to reduce double penalty issue in weather forecasting. Having this loss function added could be useful for lots of models.

**Comment by aniket2405** [comment]
Hey @jacobbieker 
While I'm going through the paper and implementation details for #52,
I want to try my hands on this enhancement too.

It seems, however, that the code is not yet available open-source. Is there any way we can access the code or do we have to wait for full publication? Thanks.

```
The GraphCast training code used in this work will be made available upon full publication of this paper. In the
meantime, the corresponding author would welcome any inquiries.
```

**Comment by jacobbieker** [comment]
Hi,

I would recommend reaching out to the authors, or just implementing it based off the paper itself.

**Comment by aniket2405** [comment]
Will pick this up now!

**Comment by jacobbieker** [comment]
Yeah, go right ahead! Would be great to have that loss added.

**Comment by TejasviniChawla** [comment]
Thanks! I'll get started on this and keep you updated.

**Comment by shyama7004** [comment]
Hlo @jacobbieker, I’d like to take up this issue if no one is currently working on it. Please let me know.

**Comment by jacobbieker** [comment]
Hi @shyama7004 this one was actually implemented in #159, although your comment seems to be meant for #133? If you want to add support for ExtremeWeatherBench, that would be great! But that is a different issue from this one.
